File: C/#ALL/PADMA REDDY LAB(2nd Sem)/6.Binary Search In Array using function/binarysearch.c
#include<stdio.h>

int binary_search(int item,int a[],int n)
{
    int low=0,mid,high=n-1;

    while(low<=high)
    {
        mid=(low+high)/2;

        if(item == a[mid]) return mid;
        else if(item < a[mid])
                    high=mid-1;
        else
                    low=mid+1;
    }

    return -1;
}

void main()
{
    int item,a[20],n,res;

    printf("Enter the value of n\n");
    scanf("%d",&n);

    printf("Enter the %d elements\n",n);

    for(int i=0;i<n;i++)
    {
        scanf("%d",&a[i]);
    }

    printf("Enter the item to be searched\n");
    scanf("%d",&item);

    res=binary_search(item,a,n);

    if(res == -1)
        printf("Item not found\n");
    else
        printf("Item  found at position %d\n",res);
}

File: CodePractices/src/binarySearch.java
public class binarySearch {
    private static boolean binarySearch(int[] arr, int low, int high, int key){
        while(low <= high) {
            int mid = low + (high-low)/2;

            if(arr[mid] == key)
                return true;
            if(arr[mid] < key)
                low = mid+1;
            else
                high = mid-1;
        }

        return false;
    }
    public static void main(String[] args) {
        int[] arr = {1, 4, 12, 23, 25, 34, 45, 52};
        int key = 23;

        boolean result = binarySearch(arr, 0, arr.length-1, key);
        System.out.println(result);
    }
}
